### PR TITLE
WELD-2685 Implement Instance.Handle interface.

### PR DESCRIPTION
**Pull request by manovotn**
Requires new Weld API release with updated CDI API version.
See also https://github.com/weld/api/pull/140

### CODE REVIEW

File: tests-arquillian/src/test/java/org/jboss/weld/tests/instance/enhanced/InstanceHandleTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        assertEquals(1, sequence.size());
+        assertEquals(alpha2Id, sequence.get(0));
+
+        alpha1.destroy();
+        // Subsequent invocations are no-op
+        alpha1.destroy();
+
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(2, sequence.size());
+        assertEquals(alpha1.get().getId(), sequence.get(1));
```
Comment (manovotn) [inline tests-arquillian/src/test/java/org/jboss/weld/tests/instance/enhanced/InstanceHandleTest.java:92]: This is actually wrong - we cannot assert the ID at this point as the instance was already destroyed. We need to change the behavior to instead throw an exception when you perform `Handle.get()` in a `Handle` that destroyed its reference. Plus we need to test that (in both test variants).

File: tests-arquillian/src/test/java/org/jboss/weld/tests/instance/enhanced/WeldInstanceTest.java [hunk 2]
```diff
@@ -50,20 +49,22 @@
...
+    public void testIsResolvable(WeldClient client) {
         ActionSequence.reset();
         assertNotNull(client);
         assertTrue(client.getAlphaInstance().isResolvable());
         assertFalse(client.getBigDecimalInstance().isResolvable());
     }
 
     @Test
-    public void testGetHandler(Client client, BeanManager beanManager) {
+    public void testGetHandler(WeldClient client, BeanManager beanManager) {
```
Comment (manovotn) [inline tests-arquillian/src/test/java/org/jboss/weld/tests/instance/enhanced/WeldInstanceTest.java:67]: Same as the other comment, this needs changing.

**Review by manovotn — COMMENTED** [review]
Found an issue with the behavior and testing; I'll fix it later today.

**Review by mkouba — COMMENTED** [review]
Hm, there is also `org.jboss.weld.environment.se.test.instance.enhanced.WeldInstanceTest` that needs to be fixed.

**Comment by manovotn** [comment]
@mkouba yeah I noticed, already fixing that locally, will be pushed soon as I get green test result :)
It is actually the only test that also asserts on the ordering so I am creating a dual variant that will test both, Weld and CDI APIs.

**Comment by manovotn** [comment]
@mkouba should be all good now

**Review by mkouba — APPROVED** [review]
Looks good.

**Pull request merged**